pallets/pool-system/src/tranches.rs

```diff
@@ -696,6 +698,7 @@ where
 	where
 		F: FnMut(&Tranche<Balance, Rate, Weight, TrancheCurrency>) -> Result<R, DispatchError>,
 	{
+		// Why not self.tranches.iter().map(f)?
```

Comment:
I think all functions in `tranches.rs` that return a `Vec`, could return instead `impl Iterator` reducing a lot of code and making it handier/rusty. I can do a PR with this suggestion if you agree.

Reply:
If I have time, I make a draft to see how it looks and then choose if we want to merge it.